opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,314 @@
+ */
+typedef int (*opal_accelerator_base_module_open_handle_fn_t)(
+    void **handle, int dev_id, void **dev_ptr);
+
+/**
+ * Closes memory mapped with opal_accelerator_base_module_open_handle_fn_t().
+ *
+ * @param[IN] dev_ptr            IPC device pointer returned from
+ *                               opal_accelerator_base_module_open_handle_fn_t()
+ */
+typedef int (*opal_accelerator_base_module_close_handle_fn_t)(
+    void *dev_ptr);
+
+/**
+ * Page-locks the memory range specified by ptr and size
+ *
+ * @param[IN] ptr            Host pointer to memory to page-lock
+ * @param[IN] size           Size in bytes of the address range to page-lock in bytes
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_host_register_fn_t)(
+    void *ptr, size_t size);
+
+/**
+ * Unregisters a memory range that was registered with
+ * opal_accelerator_base_module_host_register_fn_t.
+ *
+ * @param[IN] ptr            Host pointer to memory to unregister
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_host_unregister_fn_t)(
+    void *ptr);
+
+/**
+ * Retrieves current device id for a device associated with the local process.
+ *
+ * @param[OUT] dev_id        ID of the device
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_get_device_fn_t)(
+    int *dev_id);
+
+/**
+ * Queries if a device may directly access a peer device's memory.
+ *
+ * @param[OUT] access        Returns 1 if dev1 can directly access memory on dev2
+ *                           Returns 0 if dev1 can not directly access memory on dev2
+ * @param[IN] dev1           ID of device checking if peer device memory can be accessed
+ * @param[IN] dev2           ID of peer device on which the memory allocations
+ *                           reside.
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_device_can_access_peer_fn_t(
+    int *access, int dev1, int dev2);
+
+/*
+ * the standard public API data structure
+ */
+typedef struct {
+    /* accelerator function table */
+    opal_accelerator_base_module_check_addr_fn_t check_addr;
+    opal_accelerator_base_module_memcpy_fn_t memcpy;
+    opal_accelerator_base_module_memmove_fn_t memmove;
+    opal_accelerator_base_module_malloc_fn_t malloc;
+    opal_accelerator_base_module_free_fn_t free;
+    opal_accelerator_base_module_get_address_range_fn_t get_address_range;
+
+    opal_accelerator_base_module_is_ipc_enabled_fn_t is_ipc_enabled;
+    opal_accelerator_base_module_get_handle_fn_t get_handle;
+    opal_accelerator_base_module_open_handle_fn_t open_handle;
+    opal_accelerator_base_module_close_handle_fn_t close_handle;
+
+    opal_accelerator_base_module_host_unregister_fn_t host_register;
```

Comment:
probably just a type, should this pointer be of type ``` opal_accelerator_base_module_host_register_fn_t```

Reply:
Thanks